compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1513,17 +1516,30 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         unmentioned_fields.retain(|&(_, x)| x.name != suggested_name);
                     }
                 } else if inexistent_fields.len() == 1 {
-                    let unmentioned_field = unmentioned_fields[0].1.name;
-                    err.span_suggestion_short(
-                        ident.span,
-                        &format!(
-                            "`{}` has a field named `{}`",
-                            tcx.def_path_str(variant.def_id),
-                            unmentioned_field
-                        ),
-                        unmentioned_field.to_string(),
-                        Applicability::MaybeIncorrect,
-                    );
+                    match pat_field.pat.kind {
+                        PatKind::Lit(expr)
+                            if !same_type_modulo_infer(
```

Comment:
Can you use [`FnCtxt::can_coerce`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_typeck/check/fn_ctxt/struct.FnCtxt.html#method.can_coerce) instead?

Reply:
I fixes it!